### ISSUE TITLE
Master Build is failing due to stale commit reference in go.mod

### ISSUE THREAD

**Issue by Jorropo** [filecoin-project/lotus#8393]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [X] lotus daemon - chain sync
- [X] lotus miner - mining and block production
- [X] lotus miner/worker - sealing
- [X] lotus miner - proving(WindowPoSt)
- [X] lotus miner/market - storage deal
- [X] lotus miner/market - retrieval deal
- [X] lotus miner/market - data transfer
- [X] lotus client
- [X] lotus JSON-RPC API
- [X] lotus message management (mpool)
- [X] Other

### Lotus Version

```text
N/A
```

## Tested on:

Latest master d502eeba2b580e95facd7eb1444a34e22d76fde7

## The bug

The `go.mod` file points to `github.com/filecoin-project/specs-actors/v7@v7.0.0-20211222192039-c83bea50c402` but `c83bea50c402` is not present in the `github.com/filecoin-project/specs-actors` repo.

So running `make build` fails:
```console
$ make build
... Lots of spam

go: github.com/filecoin-project/specs-storage@v0.2.1-0.20220310131636-3fe98b33e7ea requires
	github.com/filecoin-project/specs-actors/v7@v7.0.0-20211222192039-c83bea50c402: invalid version: unknown revision c83bea50c402
```

That often happen because the `go.mod` was pointing to a commit that was not merged, then got merged (with rebase type of merge changing the commit hash), then the branch is removed and after some time when github runs GC the commit is removed.

This is affecting other projects that import lotus (I got that trying to build https://github.com/application-research/estuary/ first)

### Logging Information

```text
N/A
```

### Repo Steps

You / the CI might not have it because you are downloading a cached version from https://proxy.golang.org/.
Either use your own proxy or not `GOPROXY=direct`
```bash
export GOPROXY=direct

go clean # Hopefully remove any local cache, else get a fresh OS install in a docker or rent a server for 10 minutes on whatever on demend VPS service you like

git clone https://github.com/filecoin-project/lotus
cd lotus

make build
```

**Comment by TippyFlitsUK** [comment]
Hi @Jorropo 

Thanks for the report. I will add labels and will assign it to the right team for analysis.
We might tag you again for more information.

**Comment by arajasek** [comment]
Thanks for the report. Will be fixed by #8400
